From 5b13bf4436d06b0a5de60949dbe16c6ae8603bc7 Mon Sep 17 00:00:00 2001
From: Bobby Rong <rjl931189261@126.com>
Date: Fri, 23 Jul 2021 12:14:08 +0800
Subject: [PATCH 1/3] OperatingSystemView: display PRETTY_NAME when possible

---
 src/Views/OperatingSystemView.vala | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/src/Views/OperatingSystemView.vala b/src/Views/OperatingSystemView.vala
index fdb92e7..3340133 100644
--- a/src/Views/OperatingSystemView.vala
+++ b/src/Views/OperatingSystemView.vala
@@ -68,13 +68,28 @@ public class About.OperatingSystemView : Gtk.Grid {
         logo_overlay.add (logo);
         logo_overlay.add_overlay (icon);
 
-        // Intentionally not using GLib.OsInfoKey.PRETTY_NAME here because we
-        // want more granular control over text formatting
+        // Use GLib.OsInfoKey.PRETTY_NAME only when GLib.OsInfoKey.NAME is a
+        // prefix of it so we can get more granular control over text formatting
+        var os_name = Environment.get_os_info (GLib.OsInfoKey.NAME);
+        var os_version = Environment.get_os_info (GLib.OsInfoKey.VERSION);
+        var os_pretty_name = Environment.get_os_info (GLib.OsInfoKey.PRETTY_NAME);
+
         var pretty_name = "<b>%s</b> %s".printf (
-            Environment.get_os_info (GLib.OsInfoKey.NAME),
-            Environment.get_os_info (GLib.OsInfoKey.VERSION)
+            os_name,
+            os_version
         );
 
+        if (os_pretty_name != null
+            && os_name != null
+            && os_pretty_name.length > os_name.length
+            && os_pretty_name.slice(0, os_name.length) == os_name
+            && os_pretty_name.get(os_name.length) == ' ') {
+            pretty_name = "<b>%s</b>%s".printf (
+                os_name,
+                os_pretty_name.slice(os_name.length, os_pretty_name.length)
+            );
+        }
+
         var title = new Gtk.Label (pretty_name) {
             ellipsize = Pango.EllipsizeMode.END,
             margin_bottom = 12,
-- 
2.32.0

