--- a/src/core/keybindings.c
+++ a/src/core/keybindings.c
@@ -4112,14 +4112,6 @@ init_builtin_key_bindings (MetaDisplay *display)
                           META_KEYBINDING_ACTION_UNMAXIMIZE,
                           handle_unmaximize, 0);
 
-  add_builtin_keybinding (display,
-                          "toggle-shaded",
-                          common_keybindings,
-                          META_KEY_BINDING_PER_WINDOW |
-                          META_KEY_BINDING_IGNORE_AUTOREPEAT,
-                          META_KEYBINDING_ACTION_TOGGLE_SHADED,
-                          handle_toggle_shaded, 0);
-
   add_builtin_keybinding (display,
                           "minimize",
                           common_keybindings,
